src/adapters/http/index.ts

```diff
@@ -0,0 +1,109 @@
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve, reject) => {
+            const serverUrl = new URL(this.serverUrlExpanded);
+            const httpServer = this.glee.options?.websocket?.httpServer || http.createServer();
+            const asyncapiServerPort = serverUrl.port || 80;
+            const optionsPort = this.glee.options?.websocket?.port;
+            const port = optionsPort || asyncapiServerPort;
+
+            httpServer.on('request', (req, res) => {
+                this.res = res;
+                let { pathname } = new URL(req.url, serverUrl);
+
+                if (pathname.startsWith('/')) {
+                    pathname = pathname.substring(1);
+                }
+
+                if (!this.parsedAsyncAPI.channel(pathname)) {
+                    res.end('HTTP/1.1 404 Not Found\r\n\r\n');
+                    const err = new Error(`A client attempted to connect to channel ${pathname} but this channel is not defined in your AsyncAPI file.`)
+                    this.emit('error', err)
+                    return reject(err)
+                }
+
+                const {query: searchParams} = url.parse(req.url, true);
+                const httpChannelBinding = this.parsedAsyncAPI.channel(pathname).binding('http');
+
+                if (httpChannelBinding) {
+                    const { query, method } = httpChannelBinding;
+
+                    if(method) {
+                        if(req.method !== method) {
+                            this.emit('error', new Error(`${req.method} is not allowed for this url`));
+                            res.end('HTTP/1/1 400 Bad Request\r\n\r\n');
+                            return;
+                        }
+                    }
+
+                    if (query) {
+                        const {isValid, humanReadableError, errors} = validateData(searchParams, query);
+                        if(!isValid){
+                            const err = new GleeError({humanReadableError, errors});
+                            console.log('ERROR', err);
+                            this.emit('error', err);
+                            res.end('HTTP/1.1 400 Bad Request\r\n\r\n');
+                            return;
+                        }
+                    }
+                }
+
+                this.emit('connect', {
+                    name: this.name(),
+                    adapter: this,
+                    connection: http,
+                    channel: pathname
+                })
+
+                const msg = this._createMessage(pathname, searchParams);
+                this.emit('message', msg, http);
+
+            })
+
+
+
+            httpServer.listen(port);
+
+            this.emit('server:ready', { name: this.name(), adapter: this });
+
+            resolve(this);
+        })
+    }
+
+    async _send(message: GleeMessage): Promise<void> {
+        this.res.write(message.payload);
+        this.res.end();
```

Comment:
We should conditionally `end` the connection here. I mean, usually yeah, you close it but when you're doing Server-Sent Events (SSE), you don't want to close the connection but leave it open and let the client close it when it's done. It may need some extra work on the HTTP binding or checking the HTTP headers to see if it contains the necessary headers to do SSE.

Reply:
so should i check for headers before closing the connection?